src/test/java/com/google/devtools/build/lib/remote/util/IntegrationTestUtils.java

```diff
@@ -24,128 +23,152 @@
 import com.google.devtools.build.lib.util.OS;
 import com.google.devtools.build.lib.vfs.PathFragment;
 import com.google.devtools.build.runfiles.Runfiles;
-import java.io.File;
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.net.InetSocketAddress;
 import java.nio.channels.SocketChannel;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Comparator;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.List;
 import javax.annotation.Nullable;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
 
 /** Integration test utilities. */
 public final class IntegrationTestUtils {
   private IntegrationTestUtils() {}
 
-  private static final PathFragment WORKER_PATH =
-      PathFragment.create(
-          "io_bazel/src/tools/remote/worker"
-              + (OS.getCurrent() == OS.WINDOWS ? ".exe" : ""));
+  private static final String WORKER_RLOCATIONPATH =
+      "io_bazel/src/tools/remote/worker" + (OS.getCurrent() == OS.WINDOWS ? ".exe" : "");
 
-  private static final AtomicInteger WORKER_COUNTER = new AtomicInteger(0);
-
-  public static WorkerInstance startWorker() throws IOException, InterruptedException {
-    return startWorker(/* useHttp= */ false);
+  /**
+   * Manages a remote worker instance as a {@link TestRule}.
+   *
+   * <p>Should be kept in a static variable annotated with both {@link org.junit.ClassRule} and
+   * {@link org.junit.Rule}.
+   */
+  public static WorkerInstance createWorker() {
+    return createWorker(/* useHttp= */ false);
   }
 
-  public static WorkerInstance startWorker(boolean useHttp)
-      throws IOException, InterruptedException {
-    PathFragment testTmpDir = PathFragment.create(tmpDirFile().getAbsolutePath());
-    PathFragment stdPath = testTmpDir.getRelative("remote.std");
-    PathFragment workPath = testTmpDir.getRelative("remote.work_path");
-    PathFragment casPath = testTmpDir.getRelative("remote.cas_path");
-    int workerPort = FreePortFinder.pickUnusedRandomPort();
-    var worker =
-        new WorkerInstance(WORKER_COUNTER, useHttp, workerPort, stdPath, workPath, casPath);
-    worker.start();
-    return worker;
+  /**
+   * Manages a remote worker instance as a {@link TestRule}.
+   *
+   * <p>Should be kept in a static variable annotated with both {@link org.junit.ClassRule} and
+   * {@link org.junit.Rule}.
+   */
+  public static WorkerInstance createWorker(boolean useHttp) {
```

Comment:
@coeuvre In Bazel, this parameter is always `false`. Is it used at Google or should I inline this function?

Reply:
I think it is there primarily because we wanted to add integration tests for http cache in java. Let's keep it in case we have time to do it in the future.